### DISCUSSION TITLE
Security in production environments

### DISCUSSION THREAD

**Discussion by MyNameIsOka** [planetscale/discussion#207] [Q&A] [answered]
Hi community,

I was wondering about the security of Planetscale in production environments.
We are currently using AWS RDB which is only accessible from within a specific VPC in which the backend resides.
This doesn't seem to be possible to do in Planetscale.
What is the recommended way to prevent e.g. a brute-force attack?
Or is there maybe a way to make an IP address restriction?

Best regards,
Oka

**Comment by jgreet** [comment]
@MyNameIsOka, IP address restriction isn't currently an option on PlanetScale although it's a feature we're considering going forward.

For [Managed Cloud](https://planetscale.com/blog/introducing-planetscale-managed-cloud) and certain other enterprise accounts, we can restrict database access to your own VPCs using [AWS PrivateLink](https://aws.amazon.com/privatelink/).

When it comes to brute-force password guessing, I think it's worth noting that connection string credentials on PlanetScale are not user supplied. It isn't possible to pick a low entropy database password.

**Reply by MyNameIsOka** [reply]
Sorry for my late reply! Thank you for your answer. 
We will think about Managed Cloud but I think it's not feasible for us at this point.

**Reply by wnelson03** [reply]
@jgreet any updates on IP address restriction? This would be an extremely useful feature in PlanetScale.

**Comment by EmmanuelAmodu** [comment]
Any update or timeline for when this will be implemented? I feel naked not having any sort of network level security.

**Reply by samlambert** [reply]
https://planetscale.com/blog/introducing-IP-restrictions

IP restrictions shipped this week.

**Comment by mscoutermarsh** [comment, marked as answer]
IP restrictions are now available!

https://planetscale.com/blog/introducing-IP-restrictions
